### ISSUE TITLE
lens distortion causes bowing in orthophotos

### ISSUE THREAD

**Issue by mcnis003** [OpenDroneMap/ODM#650]
I think I am having an issue with shadows during orthorectification.

My setup is a multispectral drone that captures RGB, 655nm, 725nm, and 800nm images with 4 different cameras for agricultural research. I produce 4 separate orthophotos and align using GCPs. I have used ODM for many day's worth of data and I notice the all my photos align perfectly except the 655nm photos. The 655nm orthophotos always have a "bowing" while the other photos are perfect. I am not very familiar with the orthorectification process, but i wonder if this could be due to the shadows in my photos.

I am located in Minnesota and even at solar noon in the summer the sun is in the Southern sky. This means that my photos nearly always have a small shadow to the North. I believe this is causing a problem with the 655nm photos because there is little difference in Dn values between the shadows and the plant since plants absorb most 655nm light. Am I thinking correctly about this problem? Is there a possible fix?

655nm (with bowing):

![odm_orthophoto_cropped](https://user-images.githubusercontent.com/22203208/29008073-110e9a02-7ad6-11e7-986d-2e5e1c3c05c9.jpg)

RGB (perfect):

![odm_orthophoto_cropped](https://user-images.githubusercontent.com/22203208/29008076-1f8b061a-7ad6-11e7-8940-54dbc38b1083.jpg)

**Comment by KommandorKeen** [comment]
Looks more like a timing issue than a shadow issue or a it could be lens calibration distortion issue. Can you share the 655nm data set? 
Simon

**Comment by mcnis003** [comment]
here it is:

https://drive.google.com/open?id=0B_c8yadvyVBlbkRLLUMtenMyWmM

I know that the photos for each camera are not taken at the same time. The data in the image_metadata file has what I think is the "actual" lat lon data for each photo. The exif data and the data in the pix4d.csv file are the same for each "set" of photos. Is that what you mean by timing? I can try using that data and report back tomorrow.

update: Just double checked and I was incorrect. The drone records the same coordinates of each photo. I do have access to more precise coordinates recorded in .csv files.

**Comment by mcnis003** [comment]
My drone produces a file that records the gps location every couple milliseconds. Using this data I changed the GPS location in the image exif data. Unfortunately this has not fixed the problem.

![odm_orthophoto_cropped](https://user-images.githubusercontent.com/22203208/29051784-c6718966-7baa-11e7-9833-a118d437ae9a.png)

The other photos processed perfectly like they always have. Only the 655nm photo had a problem.

**Comment by KommandorKeen** [comment]
Agisoft makes the data bow out in the middle and ODM makes it bow in in the  middle. This is pointing to a lens distortion issue. Not much more to say at the moment.

**Comment by KommandorKeen** [comment]
OR there is a timing issue with this camera. Will look at this further....

**Comment by mcnis003** [comment]
I tried processing just the first half of the data. There are multiple trials in my photos so processing this way is not too much of a burden. I notice that this processes better when compared to a normal RGB photo from another flight.

![partial_field](https://user-images.githubusercontent.com/22203208/29074584-73f9587e-7c15-11e7-932f-438ff13d4e4f.png)

![whole_field](https://user-images.githubusercontent.com/22203208/29074589-79221f48-7c15-11e7-8db1-fa0bf0de5f40.png)

Is this consistent with a lens distortion or timing problem?

**Comment by KommandorKeen** [comment]
Not sure exactly how it all comes together in ODM, but I note that your two survey lines at the ends were in the same direction (North) and that the bow in the data is to the North at both ends. This points me towards a timing problem, but it is more gut feel than anything. Drop just one survey line from one end of the dataset and see if the result is straight, but skewed north at 1 end and south at the other. That would nail it a s a timing problem.

**Comment by mcnis003** [comment]
I have solved this problem. I believe this is cause by distortion on the right side of my 655_nm images. If you look at the images, objects on the right side of the image appear to be out of focus:

object on left side of image:

![left](https://user-images.githubusercontent.com/22203208/29242761-58b74c38-7f58-11e7-9230-d9d9f4b71260.jpg)

same object on right side of image:

![right](https://user-images.githubusercontent.com/22203208/29242763-5e83a3fa-7f58-11e7-8e26-2d5a79499593.jpg)

This problem become progressively worse as the objects move to the right side of the photo. This problem does not appear in the photos for my other cameras.

I attempted to resolve this issue by cropping my photos before ODM. In my case I cropped from 1248x950 to 1000x950. I also had to change the rescaling factor for the GCPs for the 655nm photos, but I did not need to remove GCPs that fall out of the frame of the cropped photo. The result:

![odm_orthophoto_cropped](https://user-images.githubusercontent.com/22203208/29242804-136b29aa-7f59-11e7-8741-896cce854ed4.png)

This appears to be a problem with my camera and not an issue with ODM. I will change the subject of the title to accurately reflect the issue I found.
